scene/2d/animated_sprite_2d.cpp

```diff
@@ -320,9 +321,14 @@ int AnimatedSprite2D::get_frame() const {
 }
 
 void AnimatedSprite2D::set_speed_scale(double p_speed_scale) {
+	if (speed_scale == p_speed_scale) {
+		return;
+	}
+
 	double elapsed = _get_frame_duration() - timeout;
 
-	speed_scale = MAX(p_speed_scale, 0.0f);
+	speed_scale = p_speed_scale;
+	backwards = (speed_scale < 0.0f);
```

Comment:
This is a thing I've noticed and it bothered me really bad as well, but I didn't touch to avoid breaking something. I can gladly fix it if necessary _(and backport it to 3.x because it is the same there, too)_.

Reply:
Use double literals with doubles, so it should be `0.0`. However, the compiler will auto-cast `0.0f` to `0.0` when necessary, so there's no behavior difference, and use the `f` literal when using `real_t`. Except: Anything that isn't a power of 2, like `0.1`, will have a different value with the float or double literal, so you must pay attention.

Reply:
I suppose it's no issue, then. Although, a little smelly.